pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
@smarterclayton I don't think O(hundreds) or even O(thousands) is out of line with reality.  if someone really wants to scale and finds that 4000 1-core jobs is better than 500 8-core jobs, who are we to say that's wrong.  Lots of stuff will have to be smoothed out to get that far, of course :)

To Brian, I ACK the eventual need, but I don't want to trailblaze this area as a predicate to getting v1b3 out.